src/main/java/com/uid2/admin/salt/SaltRotation.java

```diff
@@ -99,21 +104,24 @@ private boolean isRefreshable(TargetDate targetDate, SaltEntry salt) {
     }
 
     private SaltEntry[] rotateSalts(SaltEntry[] oldSalts, List<SaltEntry> saltsToRotate, TargetDate targetDate) throws Exception {
+        var keyIdGenerator = new KeyIdGenerator(oldSalts);
         var saltIdsToRotate = saltsToRotate.stream().map(SaltEntry::id).collect(Collectors.toSet());
 
         var updatedSalts = new SaltEntry[oldSalts.length];
         for (int i = 0; i < oldSalts.length; i++) {
             var shouldRotate = saltIdsToRotate.contains(oldSalts[i].id());
-            updatedSalts[i] = updateSalt(oldSalts[i], targetDate, shouldRotate);
+            updatedSalts[i] = updateSalt(oldSalts[i], targetDate, shouldRotate, keyIdGenerator);
         }
         return updatedSalts;
     }
 
-    private SaltEntry updateSalt(SaltEntry oldSalt, TargetDate targetDate, boolean shouldRotate) throws Exception {
-        var currentSalt = shouldRotate ? this.keyGenerator.generateRandomKeyString(32) : oldSalt.currentSalt();
+    private SaltEntry updateSalt(SaltEntry oldSalt, TargetDate targetDate, boolean shouldRotate, KeyIdGenerator keyIdGenerator) throws Exception {
         var lastUpdated = shouldRotate ? targetDate.asEpochMs() : oldSalt.lastUpdated();
         var refreshFrom = calculateRefreshFrom(oldSalt, targetDate);
+        var currentSalt = calculateCurrentSalt(oldSalt, shouldRotate);
         var previousSalt = calculatePreviousSalt(oldSalt, shouldRotate, targetDate);
+        var currentKey = calculateCurrentKey(oldSalt, shouldRotate, keyIdGenerator);
```

Comment:
Can we rename these and other related vars to something like "keySalt"? I keep forgetting this includes both the encryption key and encryption salt